### PR TITLE
Wt 3 my feed onboarding

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- New onboarding flow for My feed modal
- Tracking events added to track the auto flow

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-3 #done

### CODE REVIEW

**Comment by rebelchris** [comment]
@idoshamun tagging you for the analytics part, but feel free to spot other things as well 🙏

**Comment by idoshamun** [comment]
@rebelchris waiting for someone else to approve and then I'll give everything a go :)

File: packages/shared/src/components/Icon.tsx [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-type Size = 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge';
+type Size = 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge' | 'xxxlarge';
 
 const IconSize: Record<Size, string> = {
   xsmall: 'w-3 h-3',
   small: 'w-5 h-5',
   medium: 'w-6 h-6',
   large: 'w-8 h-8',
   xlarge: 'w-10 h-10',
+  xxxlarge: 'w-16 h-16',
```
Comment (kirillkurko) [inline packages/shared/src/components/Icon.tsx:12]: I see we will have a conflict of sizes - my PR uses w-14 h-14 as xxxlarge (and xxlarge is also used)

File: packages/shared/src/components/MainFeedLayout.tsx [hunk 2]
```diff
@@ -168,10 +172,32 @@ export default function MainFeedLayout({
...
   const feedName = feedNameProp === 'default' ? defaultFeed : feedNameProp;
   const isMyFeed = feedName === 'my-feed';
   const propsByFeed = getPropsByFeed({ shouldShowMyFeed });
 
+  const [isFirstSession, setIsFirstSession, isSessionLoaded] =
+    usePersistentContext(FIRST_TIME_SESSION, isFirstVisit);
+
+  useEffect(() => {
+    if (user) {
+      setCreateMyFeed(false);
```
Comment (kirillkurko) [inline packages/shared/src/components/MainFeedLayout.tsx:188]: Would be nice to have batching here - but we need to update React version (we can go with unstable_batchedUpdates from core React API, but I don't really like it even though it's pretty stable)

File: packages/shared/src/components/MainFeedLayout.tsx [hunk 3]
```diff
@@ -157,6 +160,7 @@ export default function MainFeedLayout({
     setHidden,
   } = useDynamicLoadedAnimation();
   const [createMyFeed, setCreateMyFeed] = useState(false);
+  const [myFeedMode, setMyFeedMode] = useState('manual');
```
Comment (kirillkurko) [inline packages/shared/src/components/MainFeedLayout.tsx:163]: What about enum like `FeedMode`, so we don't rely on hard concrete string values

**Review by kirillkurko — APPROVED** [review]
Looks good, left just a few minor comments more as a side-note

File: packages/shared/src/contexts/AuthContext.tsx [hunk 4]
```diff
@@ -79,6 +80,7 @@ export const AuthContextProvider = ({
   const authContext: AuthContextData = useMemo(
     () => ({
       user: user && 'providers' in user ? user : null,
+      isFirstVisit: true,
```
Comment (sshanzel) [inline packages/shared/src/contexts/AuthContext.tsx:83]: Can't remember exactly, but, won't this value be coming from boot, right?

**Review by sshanzel — APPROVED** [review]
Just had 1 question but all looks good to me 🚢

File: packages/shared/src/components/Icon.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/Icon.tsx:12]: Do you think I can safely add one bigger size?

File: packages/shared/src/components/MainFeedLayout.tsx (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/components/MainFeedLayout.tsx:188]: React version is a limitation of the extension at this point.

File: packages/shared/src/contexts/AuthContext.tsx (hunk 4 above)
Comment (rebelchris) [reply packages/shared/src/contexts/AuthContext.tsx:83]: Oh yes, wow great catch!
I overwritten this locally 🤯 (Should have been reverted back)

File: packages/shared/src/components/Icon.tsx (hunk 1 above)
Comment (kirillkurko) [reply packages/shared/src/components/Icon.tsx:12]: Yes xxxxlarge is completely safe, but name becomes a little bit weird (hope we are not going to add more sizes in the future)

Comment (rebelchris) [reply packages/shared/src/components/Icon.tsx]: We can raise it with product after this to see if we can merge the xxx and xxxx variants as they are probably meant to be the same 👀

I'll raise it while still pushing this.

**Review by kirillkurko — APPROVED** [review]
I think we all good and you can merge it

File: packages/shared/src/components/Icon.tsx (hunk 1 above)
Comment (kirillkurko) [reply packages/shared/src/components/Icon.tsx]: Sounds good, let's do this

File: packages/shared/src/components/modals/CreateMyFeedModal.tsx [hunk 5]
```diff
@@ -7,36 +7,122 @@ import TagsFilter from '../filters/TagsFilter';
...
+}: CreateMyFeedModalProps): ReactElement {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [showIntro, setShowIntro] = useState<boolean>(
+    introVariants.includes(version),
+  );
+
+  useEffect(() => {
+    trackEvent({
+      event_name: 'impression',
+      target_type: 'create my feed',
```
Comment (idoshamun) [inline packages/shared/src/components/modals/CreateMyFeedModal.tsx:77]:
```suggestion
      target_type: 'my feed onboarding',
```

File: packages/shared/src/components/modals/CreateMyFeedModal.tsx [hunk 6]
```diff
@@ -7,36 +7,122 @@ import TagsFilter from '../filters/TagsFilter';
...
+      target_id: 'modal',
+      extra: JSON.stringify({
+        origin: mode,
+        version,
+      }),
+    });
```
Comment (idoshamun) [inline packages/shared/src/components/modals/CreateMyFeedModal.tsx:83]:
```suggestion
    trackEvent({
      event_name: 'impression',
      target_type: 'my feed modal',
      target_id: version,
      extra: JSON.stringify({
        origin: mode,
      }),
    });
```

**Review by idoshamun — APPROVED** [review]
Improvement for the analytics naming. Once merged please add it to the taxonomy in the right place :)

**Pull request merged**